File: extensions/aether-forge/kotlin/src/main/kotlin/forge/ForgeStats.kt
package forge

import java.time.Instant

data class ForgeStats(
    val totalEntries: Long = 0L,
    val totalSizeBytes: Long = 0L,
    val totalDownloads: Long = 0L,
    val uniqueEcosystems: Int = 0,
    val uniquePackages: Long = 0L,
    val verifiedPackages: Long = 0L,
    val hitRate: Double = 0.0,
    val missRate: Double = 0.0,
    val averagePackageSizeBytes: Long = 0L,
    val peakDownloadsPerHour: Long = 0L,
    val activePeers: Int = 0,
    val lastUpdated: Instant = Instant.now(),
    val ecosystemBreakdown: Map<String, Long> = emptyMap()
)

File: extensions/aether-forge/swift/Sources/AetherMeshForge/ForgeStats.swift
import Foundation

public struct ForgeStats: Codable, Equatable {
    public var totalEntries: Int64
    public var totalSizeBytes: Int64
    public var totalDownloads: Int64
    public var uniqueEcosystems: Int
    public var uniquePackages: Int64
    public var verifiedPackages: Int64
    public var hitRate: Double
    public var missRate: Double
    public var averagePackageSizeBytes: Int64
    public var peakDownloadsPerHour: Int64
    public var activePeers: Int
    public let lastUpdated: Date
    public var ecosystemBreakdown: [String: Int64]

    public init(
        totalEntries: Int64 = 0,
        totalSizeBytes: Int64 = 0,
        totalDownloads: Int64 = 0,
        uniqueEcosystems: Int = 0,
        uniquePackages: Int64 = 0,
        verifiedPackages: Int64 = 0,
        hitRate: Double = 0.0,
        missRate: Double = 0.0,
        averagePackageSizeBytes: Int64 = 0,
        peakDownloadsPerHour: Int64 = 0,
        activePeers: Int = 0,
        lastUpdated: Date = Date(),
        ecosystemBreakdown: [String: Int64] = [:]
    ) {
        self.totalEntries = totalEntries
        self.totalSizeBytes = totalSizeBytes
        self.totalDownloads = totalDownloads
        self.uniqueEcosystems = uniqueEcosystems
        self.uniquePackages = uniquePackages
        self.verifiedPackages = verifiedPackages
        self.hitRate = hitRate
        self.missRate = missRate
        self.averagePackageSizeBytes = averagePackageSizeBytes
        self.peakDownloadsPerHour = peakDownloadsPerHour
        self.activePeers = activePeers
        self.lastUpdated = lastUpdated
        self.ecosystemBreakdown = ecosystemBreakdown
    }
}
